components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -8000,44 +8002,91 @@ public Comment getComment(ApiTypeWrapper apiTypeWrapper, String commentId, Integ
+            prepStmtForCount.setString(1, APIUtil.replaceEmailDomainBack(identifier.getProviderName()));
+            prepStmtForCount.setString(2, identifier.getName());
+            prepStmtForCount.setString(3, identifier.getVersion());
+            if (parentCommentID != null){
+                prepStmtForCount.setString(4, parentCommentID);
+            }
+            try (ResultSet resultSetForCount = prepStmtForCount.executeQuery()) {
+                while (resultSetForCount.next()) {
+                    total = resultSetForCount.getInt("COMMENT_COUNT");
+                }
+                if (total > 0) {
+                    if (parentCommentID == null) {
+                        sqlQuery = SQLConstants.GET_ROOT_COMMENTS_SQL;
+                    } else {
+                        sqlQuery = SQLConstants.GET_REPLIES_SQL;
+                    }
+                    try (PreparedStatement prepStmt = connection.prepareStatement(sqlQuery)) {
+                        prepStmt.setString(1, APIUtil.replaceEmailDomainBack(identifier.getProviderName()));
+                        prepStmt.setString(2, identifier.getName());
+                        prepStmt.setString(3, identifier.getVersion());
+                        if (parentCommentID != null) {
+                            prepStmt.setString(4, parentCommentID);
+                            // temporary commented below two lines as mssql doesn't support limit and offset
+//                            prepStmt.setInt(5, limit);
+//                            prepStmt.setInt(6, offset);
+                        } else {
+                            // temporary commented below two lines as mssql doesn't support limit and offset
+//                            prepStmt.setInt(4, limit);
+//                            prepStmt.setInt(5, offset);
+                        }
+                        try (ResultSet resultSet = prepStmt.executeQuery()) {
+                            while (resultSet.next()) {
+                                Comment comment = new Comment();
+                                comment.setId(resultSet.getString("COMMENT_ID"));
+                                comment.setText(resultSet.getString("COMMENT_TEXT"));
+                                comment.setUser(resultSet.getString("CREATED_BY"));
+                                comment.setCreatedTime(resultSet.getTimestamp("CREATED_TIME"));
+                                comment.setUpdatedTime(resultSet.getTimestamp("UPDATED_TIME"));
+                                comment.setApiId(resultSet.getString("API_ID"));
+                                comment.setParentCommentID(resultSet.getString("PARENT_COMMENT_ID"));
+                                comment.setEntryPoint(resultSet.getString("ENTRY_POINT"));
+                                comment.setCategory(resultSet.getString("CATEGORY"));
+                                comment.setReplies(getComments(identifier, resultSet.getString("COMMENT_ID"), 3, 0,
```

Comment:
Shall we move this to a constant? Its better to move this to a configuration. Can address separately.

Reply:
Also I think we don't need to call this when parentCommentID is already set. Shall we filter it using an if block?